spec.html

```diff
@@ -34948,8 +34950,86 @@ <h2>Syntax</h2>
           [+UnicodeMode] `-`
           CharacterClassEscape[?UnicodeMode]
           CharacterEscape[?UnicodeMode]
+
+        ClassSetExpression ::
+          ClassUnion
+          ClassIntersection
+          ClassSubtraction
+
+        ClassUnion ::
+          ClassSetRange ClassUnion?
+          ClassSetOperand ClassUnion?
+
+        ClassIntersection ::
+          ClassSetOperand `&amp;&amp;` [lookahead != `&amp;`] ClassSetOperand
+          ClassIntersection `&amp;&amp;` [lookahead != `&amp;`] ClassSetOperand
+
+        ClassSubtraction ::
+          ClassSetOperand `--` ClassSetOperand
+          ClassSubtraction `--` ClassSetOperand
+
+        ClassSetRange ::
+          ClassSetCharacter `-` ClassSetCharacter
       </emu-grammar>
+      <ul>
+        <li>
+          It is a Syntax Error if the CharacterValue of the first |ClassSetCharacter| is larger than the CharacterValue of the second |ClassSetCharacter|.
+        </li>
+      </ul>
+      <emu-grammar type="definition">
+        ClassSetOperand ::
+          ClassSetCharacter
+          ClassStringDisjunction
+          NestedClass
+
+        NestedClass ::
+          `[` [lookahead != `^`] ClassContents[+UnicodeMode, +UnicodeSetsMode] `]`
+          `[^` ClassContents[+UnicodeMode, +UnicodeSetsMode] `]`
+          `\` CharacterClassEscape[+UnicodeMode]
+      </emu-grammar>
+      <emu-note>
+        <p>The first two lines here are equivalent to CharacterClass.</p>
+      </emu-note>
+      <emu-grammar type="definition">
+        ClassStringDisjunction ::
+          `\q{` ClassStringDisjunctionContents `}`
 
+        ClassStringDisjunctionContents ::
+          ClassString
+          ClassString `|` ClassStringDisjunctionContents
+
+        ClassString ::
+          [empty]
+          NonEmptyClassString
+
+        NonEmptyClassString ::
+          ClassSetCharacter NonEmptyClassString?
+
+        ClassSetCharacter ::
+          [lookahead &notin; ClassSetReservedDoublePunctuator] SourceCharacter but not ClassSetSyntaxCharacter
+          `\` CharacterEscape[+UnicodeMode]
+          `\` ClassSetReservedPunctuator
+          `\b`
+
+        ClassSetReservedDoublePunctuator :: one of
+          `&amp;&amp;` `!!` `##`
+          `$$` `%%` `**`
+          `++` `,,` `..`
+          `::` `;;` `&lt;&lt;`
+          `==` `&gt;&gt;` `??`
+          `@@` `^^` `&grave;&grave;`
+          `~~`
```

Comment:
`esmeta` complains about this line:

```
========================================
 extract phase
----------------------------------------
[ESMeta v0.1.0] [Parser (List[spec.Production])] [28.12] failure: end of input expected

          `~~`
           ^
```

I don’t really see what’s wrong with it though. Any suggestions?

Reply:
No idea what's up with that one; probably an esmeta bug. We'll turn the check off or fix esmeta before merging; don't worry about it for now.

Reply:
I’ve filed this here: https://github.com/es-meta/esmeta/issues/145

Reply:
This has been fixed upstream, and updating the dep in this repo is now tracked here: https://github.com/tc39/ecma262/pull/3078